### PR TITLE
Center Module clean up

### PR DESCRIPTION
**Pull request by Darkeye14**
Fixes #Issue_Number

The required changes have been made.
The commits have been squashed.

Please Add Screenshots If there are any UI changes.

https://github.com/user-attachments/assets/51c9c4ef-0c3e-456a-bee6-52024d9b748c

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by Darkeye14** [comment]
The conflict can be ignored as in a PR that got merged previously, a section of the code was commented out, that fragment has been removed completely here without any effect to the working.

**Comment by therajanmaurya** [comment]
Fix conflicts

File: feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt [hunk 1]
```diff
@@ -149,8 +153,9 @@ fun CenterListScreen(
                     actions = {
                         FilledTonalButton(
                             onClick = {
+                                sync.value = true
                                 syncClicked(selectedItems.toList())
```
Comment (itsPronay) [inline feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt:157]: we are not using it anymore. Remove it. 
line no 114 as well

File: feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt [hunk 2]
```diff
@@ -149,8 +153,9 @@ fun CenterListScreen(
...
-                                resetSelectionMode()
+                             //   resetSelectionMode()
```
Comment (itsPronay) [inline feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt:158]: remove comment

File: feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt [hunk 3]
```diff
@@ -121,6 +122,9 @@ fun CenterListScreen(
         isInSelectionMode = false
         selectedItems.clear()
     }
+    val sync = remember {
```
Comment (itsPronay) [inline feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt:125]: can we use rememberSavable instead!?

File: feature/center/src/main/java/com/mifos/feature/center/navigation/CenterScreens.kt [hunk 4]
```diff
@@ -13,8 +14,8 @@ sealed class CenterScreens(val route: String) {
     data object CenterGroupListScreen : CenterScreens("center_group_list_screen/{${Constants.CENTER_ID}}") {
         fun arguments(centerId: Int) = "center_group_list_screen/${centerId}"
     }
-
+    data object SyncCenterPayloadsScreen : CenterScreens("sync_center_payloads_screen"){
+        fun arguments(centers :List<Center>) = "sync_center_payloads_screen/${centers}"
+    }
```
Comment (itsPronay) [inline feature/center/src/main/java/com/mifos/feature/center/navigation/CenterScreens.kt:19]: why did we add it? we are not using it right...

File: feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt [hunk 5]
```diff
@@ -174,14 +184,14 @@ fun SyncGroupDialogContent(
         Row(
             modifier = Modifier.fillMaxWidth()
         ) {
-            if (uiData.isSyncSuccess) {
```
Comment (itsPronay) [inline feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt:177]: why are we deleting this logic?

**Review by itsPronay — CHANGES_REQUESTED** [review]

File: feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt (hunk 1 above)
Comment (Darkeye14) [reply feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt:157]: right, ill make those necessary changes

File: feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt (hunk 5 above)
Comment (Darkeye14) [reply feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt:177]: The Cancel and hide button doesnt show and hence its implementation goes for a waste. 
Screenshot 2024-11-09 193339
Do you want me to add it back @itsPronay

**Comment by Darkeye14** [comment]
Current logic:
https://github.com/user-attachments/assets/56539c06-b80f-4993-93c5-cbffbe7630cd

Previous Working:

https://github.com/user-attachments/assets/a2a7a035-798d-48a5-b509-e06da1d77423

File: feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt (hunk 5 above)
Comment (itsPronay) [reply feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt:177]: `hide` and `cancel` should show when uiData.isSyncSuccess is false. If its not showing when syncing, the problem must be somewhere else.

Comment (itsPronay) [reply feature/center/src/main/java/com/mifos/feature/center/sync_centers_dialog/SyncCentersDialogScreen.kt:177]: keep it as it was

**Comment by Darkeye14** [comment]
Spotless has been Applied
Screenshot 2024-11-10 103541

**Comment by Darkeye14** [comment]
@therajanmaurya ji, the conflicts have been resolved

**Pull request merged**